File: Backend/entity/number.go
package entity

import "gorm.io/gorm"

type Number struct {
	gorm.Model
	Numb 			  int
	// RoleID         uint	`valid:"required~RoleID is required"`
	RoleID 			  uint  `gorm:"uniqueIndex;not null" valid:"required~RoleID is required"`
	Role              Role 	`gorm:"foreignKey:RoleID"`
}

File: Frontend/src/interfaces/Number.ts
export interface NumberRoleInterface {
    ID?: number;
    Numb?: number;
    RoleID?: number;
    Role?: {
        ID?: number;
        RoleName?: string;                 // ชื่อไทย เช่น "เจ้าของร้าน"
        RoleNickName?: "OW" | "MN" | "EMP"; // คำย่อ เช่น "OW" | "MN" | "EMP"
    } | null;
}